vat/tic.py

```diff
@@ -64,46 +64,82 @@ def __repr__(self):
     'SK': 28
 }
 
-_percent_re = re.compile('(\d*(?:\.\d*)?)%')
-
 
-def get_rates(country, date=None):
-    """Retrieve the VAT rates for the specified country.  Returns a
-       Rates object on success, or in case of error raises an exception."""
+def get_rates(country, typeVR=None):
+    """
+    Retrieve the VAT rates for the specified country.  Returns a Rates object
+    on success, or in case of error raises an exception.
+    """
+    url = str('{}{}'.format(TIC_VATRATES_HISTORY, msa_map[country]))
+    req = urllib.request.Request(url)
 
-    if date is None:
-        date = datetime.date.today()
+    response = urllib.request.urlopen(req)
 
-    req = urllib.request.Request(
-        url=TIC_VATRATESEARCH,
-        headers={'Content-Type': 'application/x-www-form-urlencoded'})
-    req.method = 'POST'
-    req.data = urllib.parse.urlencode([('listOfMsa', msa_map[country]),
-                                       ('listOfTypes', 'Standard'),
-                                       ('listOfTypes', 'Reduced'),
-                                       ('listOfTypes', 'Category'),
-                                       ('dateFilter', format_date(date))])
-
-    f = urllib.request.urlopen(req)
-
-    status = f.getcode()
+    status = response.getcode()
 
     if status != 200:
-        raise TICHTTPException(status, f.info(), f.read())
-
-    body = f.read()
-
-    xml = soupparser.fromstring(body)
-
-    row = xml.find('.//div[@id="national"]/table/tbody/tr')
-    std_rate = ''.join(row[1].itertext()).strip()
-
-    m = _percent_re.match(std_rate)
-
-    if not m:
-        raise TICException("didn't understand rate %s" % std_rate)
-
-    rate = Rate(D(m.group(1)), date)
-    rates = Rates({'Standard': rate}, {}, {})
-
-    return rates
+        raise TICHTTPException(status, response.info(), response.read())
+
+    body = response.read()
+
+    return parse_response(body, typeVR=typeVR)
+
+
+def parse_response(response, typeVR=None):
+    xml = soupparser.fromstring(response)
+
+    type_values = []
+    if typeVR is None:
+        type_values.extend(('Standard', 'Reduced'))
+    else:
+        type_values.append(typeVR)
+
+    types = {}
+    categories = {}
+
+    rows = xml.find('.//table[@id="categoriesRate"]/tbody')
+    for row in rows:
+        category = row.find('.//td[@colspan="4"]')
+        if category is not None:
+            rcategory = ''.join(category.itertext()).strip()
+            continue
+
+        for data in row:
+            if data.attrib.get('class') in type_classes:
+                text = ''.join(data.itertext()).strip()
+                if any(vrtype in text for vrtype in ['Standard', 'Reduced']):
+                    rdetail = None
+                    if '\n' in text:
+                        rtype, rdetail = text.split('\n')
```

Comment:
Co ked tu bude 2x \n ! :D No more unsafe splits
Samozrejme pokial si si ista, ze nebudu, tak asi v pohode.

Reply:
Nemalo by, ale nakoľko je to parsovanie stránky tak som to tam pre istotu obmedzila ;)